receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -269,44 +298,59 @@ func (jr *jReceiver) stopTraceReceptionLocked() error {
 	return err
 }
 
-const collectorReceiverTagValue = "jaeger-collector"
-const tchannelCollectorReceiverTagValue = "jaeger-tchannel-collector"
+func consumeTraceData(
+	ctx context.Context,
+	batches []*jaeger.Batch,
+	consumer consumer.TraceConsumer,
+) ([]*jaeger.BatchSubmitResponse, int, error) {
 
-func consumeTraceData(ctx context.Context, batches []*jaeger.Batch, consumer consumer.TraceConsumer) ([]*jaeger.BatchSubmitResponse, error) {
 	jbsr := make([]*jaeger.BatchSubmitResponse, 0, len(batches))
-
+	var consumerError error
+	numSpans := 0
 	for _, batch := range batches {
-		td, err := jaegertranslator.ThriftBatchToOCProto(batch)
-		// TODO: (@odeke-em) add this error for Jaeger observability
-		ok := false
-
-		if err == nil {
-			ok = true
-			td.SourceFormat = "jaeger"
-			consumer.ConsumeTraceData(ctx, td)
-			// We MUST unconditionally record metrics from this reception.
-			observability.RecordMetricsForTraceReceiver(ctx, len(batch.Spans), len(batch.Spans)-len(td.Spans))
+		numSpans += len(batch.Spans)
+		if consumerError != nil {
+			jbsr = append(jbsr, batchSubmitNotOkResponse)
+			continue
 		}
 
-		jbsr = append(jbsr, &jaeger.BatchSubmitResponse{
-			Ok: ok,
-		})
+		// TODO: function below never returns error, change the signature.
+		td, _ := jaegertranslator.ThriftBatchToOCProto(batch)
+		td.SourceFormat = "jaeger"
+		consumerError = consumer.ConsumeTraceData(ctx, td)
+		jsr := batchSubmitOkResponse
+		if consumerError != nil {
+			jsr = batchSubmitNotOkResponse
+		}
+		jbsr = append(jbsr, jsr)
 	}
 
-	return jbsr, nil
+	return jbsr, numSpans, consumerError
```

Comment:
we're returning only the last `consumerError` here.  Then below we pass this error into obsreport.  OBS report then uses that err value to either record numSpans as all being received or all being dropped.
```
obsreport.EndTraceDataReceiveOp(receiverCtx, span, thriftFormat, numSpans, err)
```
Should obsreport take two values?  numSpansReceived, numSpansDropped?  That way if 2 of 100 batches fail the receiver could accurately report better numbers?

Reply:
Yeah, you are right I didn't take into account Jaeger batches correctly on the method signature. Others were making this distinction mostly regarding translation and we don't want to conflate data that didn't get into the pipeline with translation issues. Let me see what can be done here.

Reply:
I'm considering that the move to custom OTLP internal format: right now this can't be resolved by doing a single call to `ConsumeTraceData` because Jaeger service information goes to the node but the new format doesn't have the issue. So we could instead put all the batches into a single call to ConsumerTraceData... 🤔

Reply:
Since the previous metric was not reporting any errors I'm siding with being slightly less wrong now and will fix this when we switch to OTLP internally.

Reply:
Opened https://github.com/open-telemetry/opentelemetry-collector/issues/579